server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/MoveProjectProfileAssociation.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.db.migrations.v52;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.persistence.Database;
+import org.sonar.server.db.migrations.*;
+import org.sonar.server.db.migrations.MassUpdate.Handler;
+import org.sonar.server.db.migrations.Select.Row;
+import org.sonar.server.db.migrations.Select.RowReader;
+
+import java.sql.SQLException;
+
+/**
+ * SonarQube 5.2
+ * SONAR-6328
+ *
+ */
+public class MoveProjectProfileAssociation extends BaseDataChange {
+
+  private static final class ProjectProfileAssociationHandler implements Handler {
+    private final Upsert setDefaultProfile;
+    private final Upsert associateProjectToProfile;
+    private final Table<String, String, String> profileKeysByLanguageThenName;
+
+    private ProjectProfileAssociationHandler(Upsert setDefaultProfile, Upsert associateProjectToProfile, Table<String, String, String> profileKeysByLanguageThenName) {
+      this.setDefaultProfile = setDefaultProfile;
+      this.associateProjectToProfile = associateProjectToProfile;
+      this.profileKeysByLanguageThenName = profileKeysByLanguageThenName;
+    }
+
+    @Override
+    public boolean handle(Row row, SqlStatement update) throws SQLException {
+      Long id = row.getLong(1);
+      String profileLanguage = extractLanguage(row.getString(2));
+      String profileName = row.getString(3);
+      Long projectId = row.getLong(4);
+      String projectUuid = row.getString(5);
+ 
+      if (profileKeysByLanguageThenName.contains(profileLanguage, profileName)) {
+        String profileKey = profileKeysByLanguageThenName.get(profileLanguage, profileName);
+
+        if (projectUuid == null) {
+          if (projectId == null || projectId == 0L) {
```

Comment:
projectId is never null because of line 57. row.getNullableLong(4) must be used.